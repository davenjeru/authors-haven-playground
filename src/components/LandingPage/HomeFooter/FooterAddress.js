import React from 'react';

const FooterAddress = () => (
  <div className="text-center">
    <h4>Authors Haven</h4>
    <p className="text-muted">July 25 2018</p>
    <p className="my-3 text-muted">
      St. Catherines, Off Thika Road, Nairobi
      <br />
    </p>
  </div>
);

export default FooterAddress;
