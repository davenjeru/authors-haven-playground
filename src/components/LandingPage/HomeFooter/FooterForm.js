import React from 'react';

const FooterForm = () => (
  <div className="text-center">
    <h5>Ask us a question</h5>
    <form className="form-row">
      <input
        type="email"
        className="form-control col-12"
        placeholder="Enter email"
      />
      <input
        type="text"
        className="form-control col-12 mt-1"
        placeholder="Question?"
      />
      <button
        type="submit"
        className="btn btn-outline-primary mx-auto mt-1 text-white"
      >
        Submit
      </button>
    </form>
  </div>
);

export default FooterForm;
