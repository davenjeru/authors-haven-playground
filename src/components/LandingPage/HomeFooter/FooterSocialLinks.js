import React from 'react';
import IconLink from '../../commons/IconLink';

const FooterSocialLinks = () => (
  <div className="text-center">
    <h5>Follow</h5>
    <IconLink
      href="https://www.instagram.com"
      iconClasses="fa fa-3x fa-instagram text-muted p-2"
    />
    <IconLink
      href="https://www.facebook.com"
      iconClasses="fa fa-facebook-square text-muted fa-3x p-2"
    />
    <IconLink
      href="https://twitter.com"
      iconClasses="fa fa-3x fa-twitter text-muted p-2"
    />
    <IconLink
      href="https://plus.google.com"
      iconClasses="fa fa-3x fa-google-plus-official text-muted p-2"
    />
  </div>
);

export default FooterSocialLinks;
