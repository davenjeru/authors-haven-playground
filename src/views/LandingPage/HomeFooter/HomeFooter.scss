.home-footer{
  position: absolute;
  width: 100%;

  hr {
    border-top-color: #6c757d;
    margin: 0 0 1rem 0;
  }
}
