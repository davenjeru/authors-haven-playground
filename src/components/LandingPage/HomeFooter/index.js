import React from 'react';
import FooterForm from './FooterForm';
import FooterSocialLinks from './FooterSocialLinks';
import FooterAddress from './FooterAddress';

const HomeFooter = () => (
  <div>
    <div className="text-md-left text-center p-4 bg-dark text-light">
      <div className="container">
        <div className="row">
          <div className="my-3 col-lg-4 col-md-6">
            <FooterAddress />
          </div>
          <div className="my-3 col-lg-4 col-md-6">
            <FooterForm />
          </div>
          <div className="my-3 col-lg-4 col-md-6">
            <FooterSocialLinks />
          </div>
          <div className="col text-center text-muted">
              Â© Copyright 2018 Authors Haven - All rights reserved.
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default HomeFooter;
