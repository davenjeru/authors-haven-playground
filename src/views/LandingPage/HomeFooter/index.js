import React from 'react';
import FooterForm from './FooterForm';
import FooterSocialLinks from './FooterSocialLinks';
import FooterAddress from './FooterAddress';
import './HomeFooter.css';

const HomeFooter = () => (
  <div className="home-footer">
    <div className="text-center p-4 bg-dark text-light">
      <div className="container">
        <div className="row">
          <div className="my-3 col-lg-4 col-md-6">
            <FooterAddress />
          </div>
          <div className="my-3 col-lg-4 col-md-6">
            <FooterForm />
          </div>
          <div className="my-3 col-lg-4 col-md-12">
            <FooterSocialLinks />
          </div>
          <div className="col-md-12 text-center text-muted">
            <hr />
              Â© Copyright 2018 Authors Haven - All rights reserved.
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default HomeFooter;
