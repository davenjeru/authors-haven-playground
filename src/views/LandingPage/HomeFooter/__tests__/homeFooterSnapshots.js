import renderer from 'react-test-renderer';
import React from 'react';
import HomeFooter from '..';

describe('The HomeFooter', () => {
  it('should not regress', () => {
    const tree = renderer.create(<HomeFooter />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
