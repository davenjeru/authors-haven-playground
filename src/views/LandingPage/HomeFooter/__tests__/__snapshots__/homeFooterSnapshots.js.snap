// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The HomeFooter should not regress 1`] = `
<div
  className="home-footer"
>
  <div
    className="text-center p-4 bg-dark text-light"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="my-3 col-lg-4 col-md-6"
        >
          <div
            className="text-center"
          >
            <h4>
              Authors Haven
            </h4>
            <p
              className="text-muted"
            >
              July 25 2018
            </p>
            <p
              className="my-3 text-muted"
            >
              St. Catherines, Off Thika Road, Nairobi
              <br />
            </p>
          </div>
        </div>
        <div
          className="my-3 col-lg-4 col-md-6"
        >
          <div
            className="text-center"
          >
            <h5>
              Ask us a question
            </h5>
            <form
              className="form-row"
            >
              <input
                className="form-control col-12"
                placeholder="Enter email"
                type="email"
              />
              <input
                className="form-control col-12 mt-1"
                placeholder="Question?"
                type="text"
              />
              <button
                className="btn btn-outline-primary mx-auto mt-1 text-white"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
        <div
          className="my-3 col-lg-4 col-md-12"
        >
          <div
            className="text-center"
          >
            <h5>
              Follow
            </h5>
            <a
              className=""
              href="https://www.instagram.com"
              target="blank"
            >
              <i
                className="fa fa-3x fa-instagram text-muted p-2"
              />
            </a>
            <a
              className=""
              href="https://www.facebook.com"
              target="blank"
            >
              <i
                className="fa fa-facebook-square text-muted fa-3x p-2"
              />
            </a>
            <a
              className=""
              href="https://twitter.com"
              target="blank"
            >
              <i
                className="fa fa-3x fa-twitter text-muted p-2"
              />
            </a>
            <a
              className=""
              href="https://plus.google.com"
              target="blank"
            >
              <i
                className="fa fa-3x fa-google-plus-official text-muted p-2"
              />
            </a>
          </div>
        </div>
        <div
          className="col-md-12 text-center text-muted"
        >
          <hr />
          Â© Copyright 2018 Authors Haven - All rights reserved.
        </div>
      </div>
    </div>
  </div>
</div>
`;
